src/util/file.c

```diff
@@ -358,6 +358,30 @@ int fileput(char *path, char *string) {
     return(0);
 }
 
+int fileputat(int fd, char *path, char *string) {
```

Comment:
I am fine with this, but just curious... What is it gaining us over using fileput(joinpath(dir1, file), ...)?